### PR TITLE
Stacktraces on blocked SafepointManager and quit if it takes too long

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

**Comment by eregon** [comment]
BTW, one easy way to test it is:
```
jt ruby -e 't=Thread.new { Truffle::Debug.dead_block }; Thread.new {sleep}; sleep 1; p t.backtrace'
```

**Comment by chrisseaton** [comment]
Maybe add that as a test then? In integration tests.

**Comment by eregon** [comment]
Test added.

File: src/main/java/org/truffleruby/language/SafepointManager.java [hunk 1]
```diff
@@ -140,10 +140,15 @@ private SafepointAction step(Node currentNode, boolean isDrivingThread) {
         return deferredAction;
     }
 
+    private static final int WAIT_TIME = 5; // seconds
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/language/SafepointManager.java]: I'd rather embed the unit in the variable name. It makes the name longer, but I think it's worth that small expense.

Comment (eregon) [reply src/main/java/org/truffleruby/language/SafepointManager.java]: Done. I wish there was a Duration class to complement TimeUnit but there is only java.time's one it seems which is much more involved and does not use TimeUnit.

**Pull request merged**